http_downloader.go

```diff
@@ -18,25 +18,61 @@ import (
 )
 
 type HttpDownloader struct {
-	Url    string
-	client *http.Client
+	Url           string
+	client        *http.Client
+	useGetForSize bool
 }
 
 func (httpDownloader HttpDownloader) GetFileInfo() (int64, bool, bool) {
-	req := httpDownloader.generateRequest("HEAD")
-	resp := httpDownloader.retryHttpRequest(req)
+	var resp *http.Response
+	var contentLength int64
+
+	if httpDownloader.useGetForSize {
+		// Use GET with Range header to determine file size
+		req := httpDownloader.generateRequest("GET")
+		req.Header.Add("Range", "bytes=0-0")
```

Comment:
can you add a link to stackoverflow where you got this idea from + can you comment that this is tested on all 3 clouds? Thx!